src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/ErrorableExtensions.cs

```diff
@@ -434,5 +434,38 @@ public static Errorable<(A, B, C, D)> With<A, B, C, D>(this Errorable<(A, B, C)>
 
             return values.Aggregate(subject, (current, v) => current.With(v).Map(applyConfiguration));
         }
+
+        /// <summary>
+        /// Executes a function returning an <see cref="Errorable{T}"/> conditionally, depending
+        /// on the result or a previous <see cref="Errorable{T}"/>.
+        /// </summary>
+        /// <typeparam name="TOld">The type of <paramref name="errorable"/> and the input to
+        /// <paramref name="whenSuccessful"/>.</typeparam>
+        /// <typeparam name="TNew">The return type of <paramref name="whenSuccessful"/></typeparam>
+        /// <param name="errorable">An existing <see cref="Errorable{T}"/> object.</param>
+        /// <param name="whenSuccessful">A function to execute on <paramref name="errorable"/> if it
+        /// was successful</param>
+        /// <returns>
+        /// An <see cref="Errorable{T}"/> containing the result of executing <paramref name="whenSuccessful"/>
+        /// if the input was sucessful, or the errors from <paramref name="errorable"/> otherwise.
+        /// </returns>
+        public static Errorable<TNew> Then<TOld, TNew>(
```

Comment:
If you want more 🎇fun✨ you can also add this as `SelectMany` and then use LINQ query syntax with `Errorable`s. (You'll need a "second" (third) argument of `Func<TOld, TNew, TFinal>`, and the result type will be `Errorable<TFinal>`.)

Reply:
Hey @Porges, isn't this, therefore, the method normally called `Bind()`?

Reply:
Whether it is, or not, I think we should move the declaration onto `Errorable<T>` and implement it for both Success and Failure.

Reply:
I normally declare them outside as extension methods to keep the "real API" smaller.

Reply:
I think "Then" works fine here as a name that makes more sense for the type. (And you can provide `SelectMany` if you want magic syntax. The problem with `Bind` is you don't gain anything!)